src/main/java/www/purple/mixxy/models/User.java

```diff
@@ -1,26 +1,54 @@
 package www.purple.mixxy.models;
 
+/**
+ * Modified by Brian_Sabz on 4/5/16.
+ * 
+ * @author Brian_Sabz
+ */
+
 import com.googlecode.objectify.annotation.Entity;
 import com.googlecode.objectify.annotation.Id;
 import com.googlecode.objectify.annotation.Index;
 
+import java.util.Date;
+
 @Entity
 @Index
 public class User {
-    
-    @Id
-    public Long id;
-    public String username;
-    public String password;
-    public String fullname;
-    public boolean isAdmin;
-
-    public User() { /* Needed by Objectify */ }
-    
-    public User(final String username, final String password, final String fullname) {
-        this.username = username;
-        this.password = password;
-        this.fullname = fullname;
-    }
- 
+
+	@Id
+	public Long id;
+	public String username;
+	public String firstname;
+	public String lastname;
+	public String email;
+
+	// @Chris - TODO: how are we going to use this map?
+	// public Map<OAuthService, String> authentications = new HashMap<>();
+
+	public Role role;
```

Comment:
@JesseTG I will look into the partial indexing for the User Roles later tonight, I tried to write up a custom condition, but it's failing on me. I will get back to you on this.